jl4-lsp/src/LSP/L4/SemanticTokens.hs

```diff
@@ -128,13 +145,20 @@ instance ToSemToken PosToken where
       , modifiers = modifiers
       }
 
--- Generic instance does not apply because we exclude the level and override
--- the token type for the name.
+-- ----------------------------------------------------------------------------
+-- Semantic Tokens for Parsed AST
+-- ----------------------------------------------------------------------------
+
+instance ToSemTokens Context PosToken (Module  Name) where
+  toSemTokens (MkModule ann _uri sects) =
+    traverseCsnWithHoles ann [toSemTokens sects]
+deriving anyclass instance ToSemTokens Context PosToken (TopDecl Name)
+
 instance ToSemTokens Context PosToken (Section Name) where
   toSemTokens (MkSection ann name maka decls) =
     traverseCsnWithHoles ann
-      [ withTokenType nameIsDirective $ toSemTokens name
-      , withTokenType nameIsDirective $ toSemTokens maka
+      [ withTokenType identIsDirective $ toSemTokens name
```

Comment:
I don't understand why it is called that. Afaiu, this doesn't check if the given identifier is a directive, it just checks whether it is an identifier.

Reply:
The `` § `var` `` the `var` is backed by a `Name` which is backed by a `TIdentifier`. To highlight this name as an `Directive`, we need to say each `TIdentifier` is to be highlighted as an `SemanticToken_Directive`.